vmdk_plugin/E2E_Tests/e2e_verifyVolmProperties_test.go

```diff
@@ -0,0 +1,133 @@
+	fmt.Println("-----Clean-up finished - current time: ", time.Now())
+}
+
+/*
+Test steps:
+1. SSH to a vm and create a volume of size 100 mb and specified disk format ["thin", "zeroedthick", "eagerzeroedthick"]
+2. Do docker inspect on the volume and verify size, disk format and attached to vm field.
+   Expected value: {"100MB", "thin/zeroedthick/eagerzeroedthick", no-value} respectively.
+3. SSH to the esx and verify the capacity, disk format and attached-to-vm field for the volume using Admin cli
+   Expected value: {"100.00MB", "thin/zeroedthick/eagerzeroedthick", "detached"} respectively.
+4. SSH to the vm and run a container and mount the volume created
+5. SSH to the vm and esx and verify the attached-to-vm field for volume - both docker cli and admin cli values should be same.
+6. Again verify capacity and disk format using docker cli and admin cli to make sure things are fine after running the container.
+*/
+func TestVolumeProperties(t *testing.T) {
+	for vmIndx := 0; vmIndx < len(vms); vmIndx++ {
+		fmt.Println("running end_to_end tests - current time: ", time.Now())
+		vmIp = vms[vmIndx]
+
+		fmt.Println("Running test on VM - ", vmIp)
+		dockerCliCheck = VolmPropsUtil.IsDockerCliCheckNeeded(vms[vmIndx])
+		for i := 0; i < len(volSizes); i++ {
+			for k := 0; k < len(formatTypes); k++ {
+				fmt.Println("Creating a volume of Format Type - ", formatTypes[k])
+				volName := TestInputParamsUtil.GetVolumeNameWithTimeStamp("dockerVol")
+				_, err := TestUtil.InvokeCommand(vms[vmIndx], "docker volume create --driver=vmdk --name="+volName+" -o size="+volSizes[i]+" -o diskformat="+formatTypes[k])
+				if err != nil {
+					fmt.Println("Failed to create a volume named: ", volName)
+					log.Fatal(err)
+				} else {
+					fmt.Println("\n successfully created a  volume -- ", volName)
+					dockerVolmRmvCmd = dockerVolmRmvCmd + " " + volName
+				}
+				volmPropertiesAdminCli := VolmPropsUtil.GetVolumePropertiesAdminCli(volName, esx)
+				expctdPropsAdmin := []string{"100.00MB", formatTypes[k], "detached"}
+				if !contains(expctdPropsAdmin, volmPropertiesAdminCli) {
+					log.Fatal("Volm properties on admin cli do not matches the expected values")
+				}
+
+				if dockerCliCheck {
+					volmPropertiesDkrCli := VolmPropsUtil.GetVolumePropertiesDockerCli(volName, vms[vmIndx])
+					expctdPropsDkr := []string{"100MB", formatTypes[k], "<no value>"}
+					if !contains(expctdPropsDkr, volmPropertiesDkrCli) {
+						log.Fatal("Volume properties on admin cli do not matches the expected values")
+					}
+				}
+				TestUtil.InvokeCommand(vms[vmIndx], "docker run -d -P --volume-driver=vmdk -v "+volName+":/vol --name e2e busybox tail -f /dev/null")
+				// Verifying attached to the vm field
+				vmNameFrmAdminCli := VolmPropsUtil.GetVmAttachedToVolUsingAdminCli(volName, esx)
+				if dockerCliCheck {
+					vmNameFrmDockerCli := VolmPropsUtil.GetVmAttachedToVolUsingDockerCli(volName, vms[vmIndx])
+					if vmNameFrmDockerCli != vmNameFrmAdminCli {
```

Comment:
need to add TODO here to extend verification by getting vmName from inventory and atleast match with either AdminCli/DockerCLI output.